### PR TITLE
Embedding Demo

### PR DESCRIPTION
**Pull request by Kaga-404**
## About The Pull Request

Experimental project related to the admin-spawnable disposable AT weapons, giving them a rare chance to embed upon impact with a softer-than-intended target. Credit to Stan for the anti-vehicle internal damage code.

## Why It's Good For The Game

This project explores some potential depth for the embedding system and throwing/dropping/volatile items, while also leading into the possibility of back-end updates for some outdated systems handling various mechanics (particularly shrapnel, embedding & embedded objects, attack/attackby/afterattack).

It is still largely a set of proof of concepts, but should be functional to serve as a rare bit of fun and creative immersive gameplay during events.

## Changelog

:cl: Kaga
add: Added in a new type of embedded shrapnel, unexploded ordnance. Currently only the event-related AT Rockets can rarely (5%) cause it when hitting unintended soft targets (human mobs) directly. Moving with one embedded has a small chance to detonate it with each step, but it can be yanked out as with other embedded objects.
add: Unexploded Anti-Tank Rockets can be manually detonated in various ways as a makeshift weapon or tool. While significantly weaker than proper rockets, they detonate in a contained direction when used. Despite the potential uses, they are highly unstable and prone to accidental detonations with each interaction. Exercise caution when handling them!
/:cl:

### CODE REVIEW

**Comment by Diegoflores31** [comment]
> ## About The Pull Request
> Experimental project related to the admin-spawnable disposable AT weapons, giving them a rare chance to embed upon impact with a softer-than-intended target. Credit to Stan for the anti-vehicle internal damage code.
> 
> ## Why It's Good For The Game
> This project explores some potential depth for the embedding system and throwing/dropping/volatile items, while also leading into the possibility of back-end updates for some outdated systems handling various mechanics (particularly shrapnel, embedding & embedded objects, attack/attackby/afterattack).
> 
> It is still largely a set of proof of concepts, but should be functional to serve as a rare bit of fun and creative immersive gameplay during events.
> 
> ## Changelog
> 🆑 Kaga add: Added in a new type of embedded shrapnel, unexploded ordnance. Currently only the event-related AT Rockets can rarely (5%) cause it when hitting unintended soft targets (human mobs) directly. Moving with one embedded has a small chance to detonate it with each step, but it can be yanked out as with other embedded objects. add: Unexploded Anti-Tank Rockets can be manually detonated in various ways as a makeshift weapon or tool. While significantly weaker than proper rockets, they detonate in a contained direction when used. Despite the potential uses, they are highly unstable and prone to accidental detonations with each interaction. Exercise caution when handling them! /🆑

very insane but if you increase the chance a lil more it would be a good deterret for not using the AT for people instead of vehicles

File: code/game/objects/items/shards.dm [hunk 1]
```diff
@@ -72,6 +72,128 @@
...
+	var/thrown = 0
+	var/cause = null
+
+/obj/item/large_shrapnel/at_rocket_dud/dropped(mob/user)
+	. = ..()
+
+	spawn(1)
+	if(!detonating && !thrown && !cause && prob(50))
+		cause = "accidental"
+		visible_message(SPAN_DANGER("You hear the click of a mechanism triggering inside [src] as [user] drops it. Uh oh."))
```
Comment (stanalbatross) [inline code/game/objects/items/shards.dm:105]:
```suggestion
		visible_message(SPAN_DANGER("You hear the click of a mechanism triggering inside \the [src] as [user] drops it. Uh oh."))
```

File: code/game/objects/items/shards.dm [hunk 2]
```diff
@@ -72,6 +72,128 @@
...
+	var/throw_channel = 2 SECONDS
+	var/detonating = 0
+	var/thrown = 0
+	var/cause = null
+
+/obj/item/large_shrapnel/at_rocket_dud/dropped(mob/user)
+	. = ..()
+
+	spawn(1)
+	if(!detonating && !thrown && !cause && prob(50))
```
Comment (stanalbatross) [inline code/game/objects/items/shards.dm:103]: make this prob(50) a var on the rocket

File: code/game/objects/items/shards.dm [hunk 3]
```diff
@@ -72,6 +72,128 @@
...
+	var/vehicle_slowdown_time = 5 SECONDS
+	var/throw_channel = 2 SECONDS
+	var/detonating = 0
+	var/thrown = 0
+	var/cause = null
+
+/obj/item/large_shrapnel/at_rocket_dud/dropped(mob/user)
+	. = ..()
+
+	spawn(1)
```
Comment (stanalbatross) [inline code/game/objects/items/shards.dm:102]: use callback timer if you're gonna do this imo

File: code/game/objects/items/shards.dm [hunk 4]
```diff
@@ -72,6 +72,128 @@
...
+/obj/item/large_shrapnel/at_rocket_dud/try_to_throw(var/mob/living/user)
+	to_chat(user, SPAN_NOTICE("You heft [src] up, preparing to throw it."))
+	user.visible_message(SPAN_DANGER("[user] strains to lift up [src]. It looks like they're trying to throw it!"))
+	throw_range = 5
+	throw_channel = 2 SECONDS
+	if(HAS_TRAIT(user, TRAIT_SUPER_STRONG))
+		throw_range = 8
+		throw_channel = 1 SECONDS
+	if(!do_after(user, throw_channel, INTERRUPT_ALL, BUSY_ICON_HOSTILE))
+		to_chat(user, SPAN_WARNING("Your attempt to throw [src] was interrupted!"))
```
Comment (stanalbatross) [inline code/game/objects/items/shards.dm:118]: \the [src]

File: code/game/objects/items/shards.dm [hunk 5]
```diff
@@ -72,6 +72,128 @@
...
+	var/mob/living/carbon/human/H = embedded_mob
+	if(H.species.flags & NO_SHRAPNEL)
+		return
+	var/obj/limb/organ = embedded_organ
+	if(istype(organ))
+		organ.take_damage(damage_on_move, 0, 0, no_limb_loss = TRUE)
+		embedded_mob.pain.apply_pain(damage_on_move)
+		if(prob(5))
+			to_chat(embedded_mob, SPAN_DANGER("The [src] sticking out of you jostles roughly against your innards! Oh no."))
+			embedded_mob.visible_message(SPAN_DANGER("The [src] sticking out of [embedded_mob] suddenly explodes!"))
```
Comment (stanalbatross) [inline code/game/objects/items/shards.dm:194]:
```suggestion
			to_chat(embedded_mob, SPAN_DANGER("\The [src] sticking out of you jostles roughly against your innards! Oh no."))
			embedded_mob.visible_message(SPAN_DANGER("\The [src] sticking out of [embedded_mob] suddenly explodes!"))
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 6]
```diff
@@ -1004,6 +1004,9 @@
 	for(var/obj/item/W in embedded_items)
 		var/obj/limb/organ = W.embedded_organ
 		// Check if shrapnel
+		if(istype(W, /obj/item/large_shrapnel))
+			var/obj/item/large_shrapnel/embedded = W
+			embedded.on_embedded_movement(src)
 		if(istype(W, /obj/item/shard/shrapnel))
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/human/human.dm:1010]:
```suggestion
		if(istype(W, /obj/item/large_shrapnel))
			var/obj/item/large_shrapnel/embedded = W
			embedded.on_embedded_movement(src)
		else if(istype(W, /obj/item/shard/shrapnel))
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**